packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -143,28 +520,50 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
    */
   readonly routeType: Protocol;
 
+  /**
+   * The gateway route action to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix or path is rewritten to '/'.
+   */
+  readonly rewrite?: HttpGatewayRouteRewrite;
+
   constructor(options: HttpGatewayRouteSpecOptions, protocol: Protocol.HTTP | Protocol.HTTP2) {
     super();
     this.routeTarget = options.routeTarget;
     this.routeType = protocol;
     this.match = options.match;
+    this.rewrite = options.rewrite;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const prefix = !isMatchPropertiesUndefined(this.match) ? this.match?.pathOrPrefix?.bind(scope).requestMatch.prefix :'/';
+
+    if (prefix && prefix[0] != '/') {
+      throw new Error(`Prefix Path must start with \'/\', got: ${prefix}`);
+    }
+    if (this.rewrite?.pathOrPrefix?.bind(scope).prefix && !prefix) {
+      throw new Error('HTTP Gateway Route Prefix Match must be set.');
     }
+    if (this.rewrite?.pathOrPrefix?.bind(scope).path && prefix) {
+      throw new Error('HTTP Gateway Route Prefix Match and Path Rewrite both cannot be set.');
+    }
```

Comment:
There is something wrong with these validations. Why cannot we make this compile-time safe?

Reply:
I thought about making it compile-time error but stumbled `rewrite` and `match` properties are defined in different level.

Do you an idea what might be a good approach on this?

Reply:
Just to highlight, we are updating `gateway-route-spec.ts` to be:

- Remove `rewrite` interface and directly put int `...SpecOptions`
- Change `GprcGatewayRouteMatch` class into the interface

Reply:
**Note:** this is still pending for change.